### PR TITLE
Migrate contributions table to react-table 8

### PR DESCRIPTION
**Pull request by amrro**
## Linear Ticket

[PRO-300 - Migrate React Table to version 8](https://linear.app/govrn/issue/PRO-300/migrate-react-table-to-version-8)

## Description
- Migrated Contribution Table to react-table v8.
- Eliminate dependency on react table's `Row` type.

## Screencaptures
All UI are technically the same, all changes are in how-it-works

### CODE REVIEW

**Comment by jonathanprozzi** [comment]
Pulling down to see if I can run the cypress tests locally

**Comment by jonathanprozzi** [comment]
Hmm ok, my recent PR (with most up to date `staging` changes) passed the Cypress tests. Wondering what's different and causing this PR to not pass CI.

**Comment by amrro** [comment]
@jonathanprozzi It's fixed now [6d92a8f](https://github.com/Govrn-HQ/govrn-monorepo/pull/334/commits/6d92a8f1dac1fbcad636850f88c161ade2710286). 
The issue was that the cypress test was looking for a component with attribute `input[title="Toggle Row Selected"]`, but can't find it. This is odd, since same UI components are used.

File: apps/protocol-frontend/src/components/ContributionsTable.tsx
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:312]: Pagination does not seem to be working for me

Comment (jonathanprozzi) [reply apps/protocol-frontend/src/components/ContributionsTable.tsx:312]: Seeing this as well

File: apps/protocol-frontend/src/components/BulkDaoAttributeModal.tsx
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/BulkDaoAttributeModal.tsx:15]: Being able to streamline the type use is huge. So much cleaner to use v8!

File: apps/protocol-frontend/src/components/ContributionsTable.tsx
Comment (jonathanprozzi) [reply apps/protocol-frontend/src/components/ContributionsTable.tsx:30]: +1 -- this is so much cleaner and more streamlined

File: apps/protocol-frontend/src/components/ContributionsTable.tsx
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:74]: So nice! Having the separate types for this led to a lot of additional complexity

File: apps/protocol-frontend/src/components/ContributionsTable.tsx
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:111]: I think this might be where we lost the pagination support -- 

The nested array makes sure we account for the newly added page sets in the parent array

File: apps/protocol-frontend/src/components/ContributionsTable.tsx
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:157]: So much cleaner !

File: apps/protocol-frontend/src/components/ContributionsTable.tsx
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:177]: Good question, This is a really cool new pattern for this

File: apps/protocol-frontend/src/components/ContributionsTable.tsx
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:275]: This is great!

**Comment by jonathanprozzi** [comment]
Looks like v8 may handle pagination and infinite scroll a bit differently: https://codesandbox.io/s/github/tanstack/table/tree/main/examples/react/virtualized-infinite-scrolling?from-embed=&file=/src/main.tsx

Overall this is amazing! I think this is good to go after we resolve infinite scroll/pagination. I'll do another round of testing before we merge, but this is so much cleaner!!

**Comment by jonathanprozzi** [comment]
Testing again, works for me! I tested with 124 contributions and it was loading groups of 20 as expected. Nice! 🚀

**Comment by twblack88** [comment]
I think this makes a little bit too many breaking changes to merge in. We'll circle back on that during planning to make a call.

**Comment by alexkeating** [comment]
> I think this makes a little bit too many breaking changes to merge in. We'll circle back on that during planning to make a call.

What breaking changes?

**Comment by jonathanprozzi** [comment]
> > I think this makes a little bit too many breaking changes to merge in. We'll circle back on that during planning to make a call.
> 
> What breaking changes?

This is referring to the issues I observed during testing I believe. I'm still seeing these -- is anyone else?

**Comment by alexkeating** [comment]
> Testing again, pagination works for me! I tested with 124 contributions and it was loading groups of 20 as expected. Nice! rocket
> 
> Having an issue with attributing to DAOs and updating contributions. Wondering if the data passed in is different than it was previously. Going to test a bit more.
> 
> It looks like it's working upon first attribute / update, but then uses a previously entered value. I'll try to add steps to reproduce and grab a screen cap.
> 
> Seeing some unexpected behavior with the Sort, but we may want to handle the sort as it's own PR. I believe we have a ticket to look into the sort inconsistencies.

These seem to be working for me now

**Review by alexkeating — APPROVED** [review]

**Pull request merged**
